### DISCUSSION TITLE
Week 13 - INCOIS hackweek

### DISCUSSION THREAD

**Discussion by eeholmes** [UW-Upwelling-Project/Summer-2023#18] [General]
@minhphan03 

Hi Minh, I could use some help getting the data from dhub to itcoocean JupyterHubs. In order of urgency:

* Can you try to convert the Zarr file to netCDF in case we need to use netCDF in R. Sadly, R and Zarr is not playing nice. Code should look something like this:
```
import xarray as xr
import zarr
fpath_zarr = "out_zarr_20140628.zarr"
zfil = xr.open_zarr()
zfil.to_netcdf()
```
* Can you move the CMIP Zarr file into the itcoocean shared-readwrite folder
* We need to get environmental variables from lat/lon/time, I have the Python code. Just need to get it working. Next post.

**Comment by eeholmes** [comment]
Here is some started code for getting environmental variables from lat/lon/time

```
# Create some points to select.  These have to be DataArray objects to do vectorized subsetting
ptime = xr.DataArray(['2023-08-09'], dims=['time'])
plat = xr.DataArray([51.5072, 48.8566, 40.7128, 35.6762], dims=['point'])  # Points to select need common dimensions
plon = xr.DataArray([0.1276, 2.3522, 74.0060, 139.6503], dims=['point'])

da.sel(time=ptime, latitude=plat, longitude=plon, method='nearest').squeeze().to_pandas()  # Squeeze to drop single time export as pandas series

df = pd.DataFrame({
    'latitude': [51.5072, 48.8566, 40.7128, 35.6762],
    'longitude': [0.1276, 2.3522, 74.0060, 139.6503],
},
    index = [1, 2, 3, 4],
                 )
df.index.name = 'points'
ds = df.to_xarray()
ds
```

Comment from my colleague:
[pandas.DataFrame.to](http://pandas.dataframe.to/)_xarray seems a more obvious way to port to xarray than xarray.Dataset.from_dataframe

```
ptime = xr.DataArray(['2023-08-03', '2023-08-09'], dims=['time'])  
plat = xr.DataArray([51.5072, 48.8566, 40.7128, 35.6762], dims=['point'])
plon = xr.DataArray([0.1276, 2.3522, 74.0060, 139.6503], dims=['point'])

da.sel(time=ptime, latitude=plat, longitude=plon, method='nearest').squeeze().to_pandas()
```

**Reply by minhphan03** [reply]
I believe we already talked about this. Look at the Get data points notebook in the python notebooks.

**Comment by eeholmes** [comment]
Oh, yeah! Thanks! I have too much on my mind. We tested the Zoom link today
so we will be using that tomorrow.
